databases/week1/sqlhomeworkweek1.sql

```diff
@@ -0,0 +1,47 @@
+-- 1. Find out how many tasks are in the task table
+SELECT COUNT(id) AS task FROM task;
+
+-- 2. Find out how many tasks in the task table do not have a valid due date
+SELECT COUNT(id) FROM task 
+WHERE due_date IS NULL;
+
+-- 3. Find all the tasks that are marked as done
+SELECT * FROM task WHERE status_id = 3;
+
+-- 4. Find all the tasks that are not marked as done
+SELECT * FROM task WHERE status_id = 0;
+
+-- 5. Get all the tasks, sorted with the most recently created first
+SELECT * FROM task 
+ORDER BY created DESC;
+
+-- 6. Get the single most recently created task
+SELECT * FROM task 
+ORDER BY created DESC
+LIMIT 1;
+
+-- 7. Get the title and due date of all tasks where the title or description contains database
+SELECT title, due_date FROM task 
+WHERE title LIKE '%database%' OR description LIKE '%database%';
+
+--  8. Get the title and status (as text) of all tasks
+SELECT t.title, s.name 
+FROM task t
+JOIN status s
+ON t.status_id = s.id; 
+
+
+-- 9. Get the name of each status, along with a count of how many tasks have that status
+SELECT s.name, COUNT(*) AS 'number of tasks'  
+FROM status s 
+LEFT JOIN task t
+ON t.status_id = s.id
+GROUP BY s.id; 
+
+-- 10. Get the names of all statuses, sorted by the status with most tasks first
+SELECT s.name, COUNT(t.status_id) 
+FROM task t
+JOIN status s
+ON t.status_id = s.id
+GROUP BY s.name
+ORDER BY COUNT(t.status_id) DESC ; 
```

Comment:
Great. Two small suggestions: Use the status id in the GROUP BY instead of the status name, just like you did in the previous query. It is much more reliable to use the primary key for grouping, when possible, since it is guarantied to always be unique. Also you could give `COUNT(t.status_id)` an alias and use that in the ORDER BY.